### PR TITLE
Task2: Inerrupting Thread

### PR DESCRIPTION
**Pull request by kristiyanpetkov**

### CODE REVIEW

File: src/com/clouway/doublethreadcounter/Demo.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+/**
+ * Created by clouway on 15-10-19.
+ */
+public class Demo {
+    public static void main(String[] args) {
+        DoubleThreadCounter firstDoubleThreadCounter = new DoubleThreadCounter("First Thread", 8);
+        DoubleThreadCounter secondDoubleThreadCounter = new DoubleThreadCounter("Second Thread", 16);
+        firstDoubleThreadCounter.start();
+        secondDoubleThreadCounter.start();
+        firstDoubleThreadCounter.setThread(secondDoubleThreadCounter);
```
Comment (mgenov) [inline src/com/clouway/doublethreadcounter/Demo.java]: Are you sure that the initialisation is performed in the right order ?

**Comment by mgenov** [comment]
Name of this PR doesn't show the intention of it.

File: src/com/clouway/doublethreadcounter/Demo.java (hunk 1 above)
Comment (kristiyanpetkov) [inline src/com/clouway/doublethreadcounter/Demo.java]: No. Invoking setThread() method should be before start() method.

Comment (kristiyanpetkov) [inline src/com/clouway/doublethreadcounter/Demo.java]: Fixed!

File: src/com/clouway/doublethreadcounter/Demo.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
+package com.clouway.doublethreadcounter;
+
+/**
+ * Created by clouway on 15-10-19.
+ */
+public class Demo {
+    public static void main(String[] args) {
+        DoubleThreadCounter firstDoubleThreadCounter = new DoubleThreadCounter("First Thread", 8);
```
Comment (mgenov) [inline src/com/clouway/doublethreadcounter/Demo.java]: Name doesn't sound good, cause you are telling that only 2 threads could be used.

Comment (kristiyanpetkov) [inline src/com/clouway/doublethreadcounter/Demo.java]: Does MultiThreadCounter sounds better?

Comment (mgenov) [inline src/com/clouway/doublethreadcounter/Demo.java]: It doesn't tell that this Thread is letting external thread to interrupt it. 

InterruptingThread sounds more reasonable.

**Comment by mgenov** [comment]
LGTM !

File: src/com/clouway/doublethreadcounter/Demo.java (hunk 2 above)
Comment (mgenov) [inline src/com/clouway/doublethreadcounter/Demo.java]: Or DocketCountdownThread

**Pull request merged**